subprojects/docs/src/docs/release/notes.md

```diff
@@ -2,6 +2,23 @@
 
 Here are the new features introduced in this Gradle release.
 
+### Dependency constraints
+
+With [dependency constraints](userguide/managing_transitive_dependencies.html#sec:dependency_constraints), Gradle adds a mechanism to express constraints over transitive dependencies. They are used during dependency resolution of a build and are also published when the Gradle module metadata format is used (see below).
+
+    dependencies {
+        implementation 'org.apache.httpcomponents:httpclient'  
+    }
+    
+    dependencies {
+        constraints {
```

Comment:
I don't quite understand what the constraint is supposed to be in this example? It looks to me as if it would be versions-specific. As a user I can use the HTTP client constraint to avoid having to declare the version explicitly. Sounds more like a recommendation to me than a constraint. I am sure you have good reasons for calling it that but the release notes don't provide a clear picture.

What does a user have to know about publishing the constraints? How does it look like? Are there limitations? Are they compatible with know repository formats?

Reply:
This is the first simple version of dependency constraints. You can look at the example as a recommendation. But it also describes the minimum needed version. So in this case it upgrades the version brought in transitively. So it is a constraint. Lower versions are now allowed.
I added an explanation to the example. Also, the docs section is already linked, which contains a similar example.
(Note: In one of the future releases we will officially support "version constraints" - many of which are already implemented. Which then allow you to define a `strict` or `reject` constraints for example. I think then the term "dependency constraint" automatically becomes more fitting.)

About publishing: In my mind, this is the "killer" features why you should use constraints instead of a resolution strategy of some sort. So I wanted to point that out here:
"That's why we introduce this new concept. If you develop a library - this is for you! (you need to wait more until this is stable, but hey there is this new experimental feature and you can try it out now)".

I could elaborate a bit more here to better get this message across. Or just remove the part about publishing. What do you think?

Reply:
> About publishing: In my mind, this is the "killer" features why you should use constraints instead of a resolution strategy of some sort. So I wanted to point that out here.

That's important to know as a user. Thanks.

Reply:
I added an explanation why publishing is important.